### PR TITLE
fix docs format on index.md

### PR DESCRIPTION
**Pull request by laisbsc**
The aim of this PR is to:
+ Correct `index.md` file to point to the right markdown files on the doctree.
+ Fix local merge conflicts on `RELEASE.md`.
+ Fix navbar references on newly deployed docs (`conf.py`).
+ Add placeholder file on [Management](docs/source/02_get_started/05_management.md), [Support](docs/source/02_get_started/07_support.md), [Infrastructure](docs/source/05_reference/01_qhub_infrastructure.md) and [Architecture](docs/source/06_developers_contrib_guide/01_architecture.md).
+ Add section on [Troubleshooting](docs/source/02_get_started/06_troubleshooting.md).

### CODE REVIEW

File: docs/source/06_developers_contrib_guide/04_local_development.md [hunk 1]
```diff
@@ -1,268 +0,0 @@
-# Testing
```
Comment (costrouc) [inline docs/source/06_developers_contrib_guide/04_local_development.md:1]: We need this file for docs on how to do local testing and develop. Is there a better place that this should be located in the docs?

**Comment by laisbsc** [comment]
The way I thought about this was a bit different: I had local and cloud deployment on the usage and install.
We should add a testing section.

I thought this was accepted when we discussed the new ToC of the docs. I'm studying for a school test atm so if you can, please just add the info there and I can try and make sense of a structure after my exam tomorrow.

File: docs/source/06_developers_contrib_guide/04_local_development.md (hunk 1 above)
Comment (laisbsc) [reply docs/source/06_developers_contrib_guide/04_local_development.md:1]: The way I thought about this was a bit different: I had local and cloud deployment on the usage and install.
If we need a testing section, we should add one :)

I thought this was accepted with this structure when we discussed the new ToC of the docs. I'm studying for a school test atm so if you can, please just add the info there and I can try and make sense of a structure after my exam tomorrow.

The info that was on that file was transferred half to the install and half to the usage docs.

**Review by costrouc — APPROVED** [review]
Looks great to me! Thanks for the fixes @laisbsc

**Pull request merged**